game-core/src/main/java/games/strategy/engine/data/RouteScripted.java

```diff
@@ -36,18 +36,10 @@ public int numberOfSteps() {
     return super.getSteps();
   }
 
-  @Override
-  public Territory getEnd() {
-    if (super.getEnd() == null) {
-      return super.getStart();
-    }
-    return super.getEnd();
-  }
-
   @Override
   public Territory getTerritoryAtStep(final int i) {
     try {
-      if (super.getEnd() == null || super.getTerritoryAtStep(i) == null) {
+      if (super.getTerritoryAtStep(i) == null) {
```

Comment:
getTerritoryAtStep has to be non-null as well, right?
So the only case this code does anything is when i is outside of the array bounds and an exception is thrown.

Reply:
Just had a look at the other part of the method, this is really ugly :(

Reply:
It was also interesting how much copy/paste there was in this clean-up, many of the null checks were clearly just copied from other parts of the code.

> getTerritoryAtStep has to be non-null as well, right?

Likely should be non-null as we disallow adding null territories. It might be easy to find and clean up as well.